test/e2e/framework/pod/wait.go

```diff
@@ -211,13 +218,83 @@ func WaitForPodsRunningReady(ctx context.Context, c clientset.Interface, ns stri
 	}))
 
 	// An error might not be fatal.
-	if err != nil && notReady >= 0 && notReady <= allowedNotReadyPods {
-		framework.Logf("Number of not-ready pods (%d) is below the allowed threshold (%d).", notReady, allowedNotReadyPods)
+	if nOk+len(otherPods) >= minPods && len(otherPods) <= allowedNotReadyPods {
```

Comment:
This is the only actual logic change I made to this function. I know that the team is hesitant to make any changes to these functions, and I'm happy to undo this change if we want.

However, I felt like this was the one change that had to be kept. Previously, this check would pass and we would return `nil` in obvious failure scenarios such as every Pod failed. Surely that's a logic bug that no test is relying upon. The entire purpose of this function is to check that "enough" Pods are Running and Ready, so hopefully is acceptable to ensure that this function actually checks that.

Reply:
I have not had the time to verify the logic of behind this change. Usually that would be a reason to not merge it. I don't doubt that you are doing the right thing, but usually it's the well-intended-but-not-fully-understood (by author and reviewer!) changes which cause regressions.

Do we really need to fix something here?

Reply:
Is there a need for fuctions in tests to do what they're supposed to? Perhaps not "need", but it does seem like a good idea! (Apologies if that sounds snarky via text, I really don't mean for it to =D). I think my largest concern I have about the current code is:

1. Many of the Pods this function is waiting on fail
2. This function is bugged so it doesn't return an error
3. Because the Pods failed, another part of the test fails
4. It's difficult to debug why the test failed

That's just hypothetical though.

This is my first K8s PR and I'm just trying to dip my toes in the water. I may have opinions I want to fight for down the road, but this definitely isn't one of them. I'm more than happy to defer to your judgement here.

Reply:
> Is there a need for functions in tests to do what they're supposed to?

Sure, but who decides what they are supposed to do? This particular example has a deceptively simple name that implies a certain purpose, but then clearly does a whole lot more than just that.

> Apologies if that sounds snarky via text

No problem :smile:

> I'm more than happy to defer to your judgement here.

I prefer to not change it. It's a simple risk assessment: no-one is complaining about this *in the context of checking the cluster readiness* (which is what this function will be used for going forward). But if we change it, we might break someone and then get complaints.

Reply:
The reimplementation of WaitForPodsRunningReady can be simpler and needs to be different - that's okay.